pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -325,6 +327,7 @@ func (p *AgentPoller) Start() {
 	for i := 0; i < len(p.workers); i++ {
 		log.Log.Infof("Starting agent poller with commands: %v", p.workers[i].AgentCommands)
 		go p.workers[i].Poll(func(commands []AgentCommand) {
+			executeApiOperations(commands, p.Connection, p.agentStore, p.domainName)
```

Comment:
I think this is a hacky way to implement this change. Can you instead modify the `PollerWorker` to support libvirt API calls? It may also help in the other PR, where the rest of the guest agent calls will be moved to libvirt API.

Reply:
Done.